src/sacn/source.c

```diff
@@ -1033,42 +1026,166 @@ int sacn_source_process_manual(void)
+ * If this call fails, the caller must call sacn_source_destroy() on all sources, because the source API may be in an
  * invalid state.
  *
  * Note that the networking reset is considered successful if it is able to successfully use any of the
  * network interfaces passed in.  This will only return #kEtcPalErrNoNetints if none of the interfaces work.
  *
- * @param[in] handle Handle to the source for which to reset the networking.
- * @param[in] universe Universe to reset netowrk interfaces for.
- * @param[in, out] netints Optional. If non-NULL, this is the list of interfaces the application wants to use, and the
- * status codes are filled in.  If NULL, all available interfaces are tried.
- * @param[in, out] num_netints Optional. The size of netints, or 0 if netints is NULL.
- * @return #kEtcPalErrOk: Source changed successfully.
+ * @param[in, out] netints If non-NULL, this is the list of interfaces the application wants to use, and the status
+ * codes are filled in.  If NULL, all available interfaces are tried.
+ * @param[in] num_netints The size of netints, or 0 if netints is NULL.
+ * @return #kEtcPalErrOk: Networking reset successfully.
  * @return #kEtcPalErrNoNetints: None of the network interfaces provided were usable by the library.
- * @return #kEtcPalErrInvalid: Invalid parameter provided.
  * @return #kEtcPalErrNotInit: Module not initialized.
- * @return #kEtcPalErrNotFound: Handle does not correspond to a valid source, or the universe was not found on this
- *                              source.
  * @return #kEtcPalErrSys: An internal library or system call error occurred.
  */
-etcpal_error_t sacn_source_reset_networking(sacn_source_t handle, uint16_t universe, SacnMcastInterface* netints,
-                                            size_t num_netints)
+etcpal_error_t sacn_source_reset_networking(SacnMcastInterface* netints, size_t num_netints)
 {
 #if SACN_SOURCE_ENABLED
-  ETCPAL_UNUSED_ARG(handle);
-  ETCPAL_UNUSED_ARG(universe);
-  ETCPAL_UNUSED_ARG(netints);
-  ETCPAL_UNUSED_ARG(num_netints);
+  etcpal_error_t result = kEtcPalErrOk;
+
+  if (!sacn_initialized())
+    result = kEtcPalErrNotInit;
+
+  if ((result == kEtcPalErrOk) && sacn_lock())
+  {
+    sacn_sockets_reset_source();
+
+    for (size_t i = 0; (result == kEtcPalErrOk) && (i < get_num_sources()); ++i)
+    {
+      SacnSource* source = get_source(i);
+
+      source->num_netints = 0;  // Clear source netints, will be reconstructed when netints are re-added.
+
+      for (size_t j = 0; (result == kEtcPalErrOk) && (j < source->num_universes); ++j)
+      {
+        SacnSourceUniverse* universe = &source->universes[j];
+
+        universe->netints.num_netints = 0;
+        result = sacn_initialize_source_netints(&universe->netints, netints, num_netints);
+
+        for (size_t k = 0; (result == kEtcPalErrOk) && (k < universe->netints.num_netints); ++k)
+          result = add_to_source_netints(source, &universe->netints.netints[k]);
```

Comment:
If you have 3 netints and the second one is bad, won't this drop the third netint as well?

Reply:
I don't think so. My intent was for sacn_initialize_source_netints to initialize universe->netints with only the netints that have an OK status. So the second netint you mention that failed would not even make it into universe->netints. Only the first and third netints would be in universe->netints. So this loop would be k = 0 and k = 1 since universe->netints.num_netints would be 2.

Reply:
Keeping in mind this was what I intended. It could very well still be breaking early, but I don't see where.

Reply:
Ah, I think the misunderstanding was on my side.  A closer look at add_to_source_netints() made me realize that the result check in the for loop was for a real error, not a bad interface.

Reply:
Yeah, there won't be an error unless all the interfaces are down.